scipy/signal/windows.py

```diff
@@ -709,6 +709,92 @@ def hann(M, sym=True):
 hanning = hann
 
 
+def tukey(M, alpha=0.5, sym=True):
+    r"""Return a Tukey window, also known as a tapered cosine window. 
+
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an empty
+        array is returned.
+    alpha : float, optional
+        Shape parameter of the Tukey window, representing the faction of the
+        window inside the cosine tapered region. 
+        If zero, the Tukey window is equivalent to a rectangular window.
+        If one, the Tukey window is equivalent to a Hann window.
+    sym : bool, optional
+        When True (default), generates a symmetric window, for use in filter
+        design.
+        When False, generates a periodic window, for use in spectral analysis.
+
+    Returns
+    -------
+    w : ndarray
+        The window, with the maximum value normalized to 1 (though the value 1
+        does not appear if `M` is even and `sym` is True).
+
+    References
+    ----------
+    .. [1] Tukey, J.W., (1967), "An introduction to the calculations of 
```

Comment:
Not personally, I took it on faith from Wikipedia... Alternatively, I was able to easily access the other reference listed there, which contains a plethora of information on various windows, including the Tukey. I'll use that one instead.